src/org/traccar/protocol/EelinkProtocolDecoder.java

```diff
@@ -209,41 +325,72 @@ protected Object decode(
 
         ChannelBuffer buf = (ChannelBuffer) msg;
 
-        buf.skipBytes(2); // header
-        int type = buf.readUnsignedByte();
-        buf.readShort(); // length
-        int index = buf.readUnsignedShort();
+        // check that incoming protocall has correct eelink prefix
+        if (buf.readShort() != HEADER_KEY) {
+            Log.warning(new IllegalArgumentException("Invalid Header"));
+            return null;
+        }
+
+        int type = buf.readUnsignedByte(); // protocol number
 
-        if (type != MSG_GPS && type != MSG_DATA) {
-            sendResponse(channel, type, index);
+        short length = buf.readShort(); // length
+        if (buf.readableBytes() != length) { // check remaining length
+            Log.warning(new IllegalArgumentException("Packet length error"));
+            return null;
         }
 
-        if (type == MSG_LOGIN) {
+        index = buf.readUnsignedShort();
+
+        // check if we have received an old packet
+        /*int newIndex = buf.readUnsignedShort(); // current message index
```

Comment:
We don't do such things in Traccar. Just respond with received index. I don't see much harm in occasional duplicate message, but it's also possible to filter it out if user doesn't want duplicates in the database.